### PR TITLE
Re-type Captain's armor to a space rig.

### LINKED ISSUE #22294: Captains armor doesnt fit
(WEB REPORT BY: alandemarest REMOTE: 206.221.180.138:7777)
# Revision
d1cd189c68cdc540d1dd0e4e1787dc170a09daea

# Description
Captains armor(chest) cant be put into suit storage unit, and therefore can't be decontaminated or stored.
# Steps to Reproduce
Take Captains armor out of storage, try to put in
# What you Expected

# What Actually Happened

### LINKED ISSUE #24557: Captains Armour
(WEB REPORT BY: salmonistasty REMOTE: 206.221.180.138:7777)
# Revision
57d78c144e6a56ca7239ab113fd7952b53da9d73

# Description

Captains armour is space proof and also spawns inside Suit Storage Unit, but it can't be put back inside it.

# Steps to Reproduce

try to put captains armour to the Suit Storage Unit

# What you Expected

to put the armour inside

# What Actually Happened

nothing

### PR DESCRIPTION
**Pull request by ghost**
[bugfix]
Fixes: https://github.com/vgstation-coders/vgstation13/issues/24557
Fixes: #22294 
Special ❄️ captains armor has been re-typed to a space rig. I also renamed it to ``Captain's rig armor``. Is that stupid? 
image

:cl:
 * tweak: Captain's armor is now an armored rigsuit.
 * tweak: Captain's SSU now has the right department colors.
 * bugfix: Fixed bug where Captain's armor could not be placed back into a SSU.

### CODE REVIEW

**Comment by DrSnips** [comment]
Why is the captain's armor even really considered armor when its more of a fancy rig/space suit with higher armor values?

**Comment by ghost** [comment]
> 
> 
> Why is the captain's armor even really considered armor when its more of a fancy rig/space suit with higher armor values?

I was gonna convert it to a spacesuit subtype but I don't feel like fucking with that.

**Comment by beesil** [comment]
resolves #22294 as well

**Comment by Probe1** [comment]
I really appreciate you writing this code but this isn't a good solution.  Please just re-type the captains armor as a RIG suit.

**Comment by DrSnips** [comment]
At this point it might as well be. That "armor" has been snowflaked to hell and back to basically be a subtype of a sec rig suit. Light helmet, hold most combat things, high armor value, atmos stats are similar enough.

**Comment by Dacendeth** [comment]
A lot of races wont be able to wear it if you turn it into a rig suit like they otherwise can now.

**Comment by Probe1** [comment]
Making one override to allow applicable races to wear it is better than the amount of code we're maintaining here that will one day break and someone will say "haha crazy old coders and their snowflaked solutions".

**Comment by ghost** [comment]
@Probe1 Yeah, good idea. I'm not done yet, I want to do custom sprites for the captain's SSU.

**Comment by SonixApache** [comment]
For the record, just add the proper species to the `species_restricted` var of a clothing item to allow that species to wear it, and add that same species to `species_fit` to make it use species-specific icons when worn.

**Comment by Intigracy** [comment]
or set species_restricted = null so all species can wear it
and still use species_fit to make species-specific icons work

(as was just done oops lmao)

File: code/modules/clothing/spacesuits/captain.dm [hunk 1]
```diff
@@ -44,6 +38,7 @@
 	armor = list(melee = 65, bullet = 50, laser = 50, energy = 25, bomb = 50, bio = 100, rad = 50)
 	siemens_coefficient = 0.7
 	heat_conductivity = SPACESUIT_HEAT_CONDUCTIVITY
+	head_type = /obj/item/clothing/head/helmet/space/rig/capspace
 
 /obj/item/clothing/suit/armor/captain/old
```
Comment (DrSnips) [inline code/modules/clothing/spacesuits/captain.dm]: Might want to make sure this thing still works as intended since its previous parent is now dead.

Comment (ghost) [reply code/modules/clothing/spacesuits/captain.dm]: I did test it a bunch of times. I had some issues so I had to just keep testing. It SHOULD be fine.

Comment (Intigracy) [reply code/modules/clothing/spacesuits/captain.dm]: So it has the right name, description, armor, etc?

**Comment by Blithering** [comment]
Doesn't the captains armour predate rigsuits codewise? Probably where all the weird coding for it came from. I reckon it should be whatever deathsquad armour is classed as, since they're supposed to be different models of the same thing.

**Comment by ghost** [comment]
> 
> 
> Doesn't the captains armour predate rigsuits codewise? Probably where all the weird coding for it came from. I reckon it should be whatever deathsquad armour is classed as, since they're supposed to be different models of the same thing.

deathsquad armor **is**  classed as a rig.

**Comment by Durgen** [comment]
Does this effect tank storage? iirc it only holds small tanks

**Comment by Probe1** [comment]
Yes it will.  @Ephememory Is this still WIP?

**Comment by ghost** [comment]
Now that it can accept large oxygen tanks, **yes it is no longer WIP.**

**Comment by Intigracy** [comment]
You didn't address /obj/item/clothing/suit/armor/captain/old
It now doesn't inherit anything except from /armor

**Comment by Probe1** [comment]
This was a journey.  And this is the end.

**Pull request merged**